packages/editor/src/utils/with-history/test/index.js

```diff
@@ -167,6 +167,62 @@ describe( 'withHistory', () => {
 		} );
 	} );
 
+	it( 'should overwrite present state with option.shouldOverwriteState right after ignored action', () => {
+		const complexCounter = ( state = { count: 0 }, action ) => {
+			if ( action.type === 'INCREMENT' ) {
+				return {
+					...state,
+					count: state.count + 1,
+				};
+			} else if ( action.type === 'IGNORE' ) {
+				return {
+					...state,
+					ignore: action.content,
+				};
+			}
+
+			return state;
+		};
+
+		const reducer = withHistory( {
+			shouldOverwriteState: ( action, previousAction ) =>
+				previousAction && action.type === previousAction.type,
```

Comment:
Pretty nitpicky but I find this a bit tougher to read; implicit returns I find a bit hard to read when they aren't one-liners or at least wrapped in parens on multiline. How about:
```suggestion
			shouldOverwriteState: ( action, previousAction ) => (
				previousAction && action.type === previousAction.type
			),
```